### ISSUE TITLE
[BUG] rich.reconfigure raises AttributeError if invoked before _console initialization

### ISSUE THREAD

**Issue by brianwisti** [willmcgugan/rich#1407]
Found the teeniest bug, which I wouldn't have come across if I wasn't being sneaky setting Rich up for a blog post.

**Read the docs**

[`rich.reconfigure`]: https://rich.readthedocs.io/en/stable/reference/init.html?highlight=reconfigure#rich.reconfigure

[`rich.reconfigure`][] docs do not match code, but that's a different issue. 

**Describe the bug**

Attempting to reconfigure the global Console on application start triggers an AttributeError.

**To Reproduce**

With a fresh checkout of the code, reproducible by testing `rich.reconfigure` in isolation:

```shell
$ poetry run pytest tests/rich_print.py -k reconfigure
= test session starts =
platform linux -- Python 3.9.1, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/random/src/rich/tests, configfile: pytest.ini
plugins: cov-2.12.1
collected 4 items / 3 deselected / 1 selected

tests/test_rich_print.py F                                                                              [100%]

= FAILURES =
_ test_reconfigure_console _

    def test_reconfigure_console():
>       rich.reconfigure(width=100)

tests/test_rich_print.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (), kwargs = {'width': 100}, Console = <class 'rich.console.Console'>
new_console = <console width=100 None>

    def reconfigure(*args: Any, **kwargs: Any) -> None:
        """Reconfigures the global console by replacing it with another.

        Args:
            console (Console): Replacement console instance.
        """
        from rich.console import Console

        new_console = Console(*args, **kwargs)
>       _console.__dict__ = new_console.__dict__
E       AttributeError: 'NoneType' object has no attribute '__dict__'

rich/__init__.py:44: AttributeError
= short test summary info =
FAILED tests/test_rich_print.py::test_reconfigure_console - AttributeError: 'NoneType' object has no attribu...
= 1 failed, 3 deselected in 0.09s =
```

*Workaround*

Access the global Console prior to `reconfigure` via printing or `rich.get_console`.

``` python
_ = rich.get_console()
rich.reconfigure(width=100)
```

*Fix*

Update `rich.reconfigure` to access `_console` via the accessor function:

```python
  def reconfigure(*args: Any, **kwargs: Any) -> None:
      """..."""
      from rich.console import Console

      new_console = Console(*args, **kwargs)
      get_console().__dict__ = new_console.__dict__
```

I can do a proper pull request if needed, but the fix was quick and small so I figured I'd drop it here while it was still at the top of my head.

**Platform**
What platform (Win/Linux/Mac) are you running on? What terminal software are you using?

*Platform*

WSL2 / Ubuntu 20.04 on Windows 10

*Terminal*

Windows Terminal Preview

**Diagnose**
I may ask you to cut and paste the output of the following commands. It may save some time if you do it now.

```
python -m rich.diagnose
python -m rich._windows
pip freeze | grep rich
```

**Did I help?**

If I was able to resolve your problem, consider [sponsoring](https://github.com/sponsors/willmcgugan) my work on Rich, or [buy me a coffee](https://ko-fi.com/willmcgugan) to say thanks.

**Comment by willmcgugan** [comment]
Possible fix: https://github.com/willmcgugan/rich/pull/1426

**Comment by willmcgugan** [comment]
Fixed in v10.8.0 Thanks
